domains/authorization/Proxy.IntegrationTests/ProxyBaseIntegrationTests.cs

```diff
@@ -0,0 +1,129 @@
+using System.Net;
+using FluentAssertions;
+using Proxy.Controllers;
+using Proxy.IntegrationTests.Setup;
+using WireMock.RequestBuilders;
+using WireMock.ResponseBuilders;
+
+namespace Proxy.IntegrationTests;
+
+public class ProxyBaseIntegrationTests(ProxyIntegrationTestFixture fixture) : IClassFixture<ProxyIntegrationTestFixture>
+{
+    private HttpClient CreateClientWithOrgIds(List<string> orgIds)
+    {
+        return fixture.Factory.CreateAuthenticatedClient(orgIds: orgIds);
+    }
+
+    [Fact]
+    public async Task Proxy_Forwards_Request_To_Downstream_With_Version_Header()
+    {
+
+        fixture.WalletWireMockServer
+            .Given(Request.Create().WithPath("/v1/certificates").UsingGet())
+            .RespondWith(Response.Create().WithStatusCode(200));
+
+        var orgIds = new List<string> { Guid.NewGuid().ToString(), Guid.NewGuid().ToString() };
+        var client = CreateClientWithOrgIds(orgIds);
+        var organizationId = orgIds[0];
+
+        var request = new HttpRequestMessage(HttpMethod.Get, $"/certificates?organizationId={organizationId}");
+        request.Headers.Add("EO_API_VERSION", ApiVersions.Version20250101);
+
+        var response = await client.SendAsync(request);
+
+        response.StatusCode.Should().Be(HttpStatusCode.OK);
+    }
+
+    // TODO: Make sure it actually follows http standards
+    // [Fact]
+    // public async Task Proxy_Returns_Bad_Gateway_When_Upstream_Returns_Invalid_Response()
+    // {
+    //     var orgIds = new List<string> { Guid.NewGuid().ToString() };
+    //     var client = CreateClientWithOrgIds(orgIds);
+    //     var organizationId = orgIds[0];
+    //
+    //     var request = new HttpRequestMessage(HttpMethod.Get, $"/certificates?organizationId={organizationId}");
+    //     request.Headers.Add("EO_API_VERSION", ApiVersions.Version20250101);
+    //
+    //     var response = await client.SendAsync(request);
+    //
+    //     response.StatusCode.Should().Be(HttpStatusCode.BadGateway);
+    // }
+
```

Comment:
This

Reply:
We proxy the responses back 100% from Wallet. We don't check their response. We want to return their validation errors etc. Otherwise we should create a new task for this feature. Not entirely sure what Request we want this to happen for? Do you have an example in mind?

Reply:
If the wallet is down, or unreachable, what should the proxy do? HTTP standard says return 502

Reply:
But we are not a proxy for externals. They shouldn't know about our internals. So whatever wallets gives us is fine or even 500 for showing something is wrong. We are not a gateway for externals. Its just implementation detail.

Reply:
How will we differentiate between a 500 error on wallet side, or 500 error on proxy side?

Reply:
The error message / logging. The consumer shoulnd't care about where error 500 is from.